tests/ttnn/unit_tests/gtests/test_multiprod_queue.cpp

```diff
@@ -52,7 +52,7 @@ TEST_F(MultiCommandQueueSingleDeviceFixture, TestMultiProducerLockBasedQueue) {
                 t0_host_data[i] = bfloat16(static_cast<float>(2 + j));
             }
             // Allocate and write buffer
-            auto t0_input_buffer = ttnn::allocate_buffer_on_device(tensor_buf_size * datum_size_bytes, device, tensor_shape, DataType::BFLOAT16, Layout::TILE, mem_cfg);
+            auto t0_input_buffer = tt::tt_metal::tensor_impl::allocate_buffer_on_device(tensor_buf_size * datum_size_bytes, device, tensor_shape, DataType::BFLOAT16, Layout::TILE, mem_cfg);
```

Comment:
yes. there is no need for that api in ttnn. 

These two methods were absolutely identical with main difference that ttnn version returned Buffer defined in ttnn namespace. Such a mess. That version of Buffer override allocate/deallocate to make it asynchronous. Now when tt::tt_metal::Buffer does it - its best to just remove those functions.

This is not the perfect state, but it leaves us in a better place till further improvements

Reply:
The original goal was to keep async at the TTNN level, which is why we actively made a choice to introduce a buffer at the TTNN level with thread-safety and asynchronous functionality baked in. This is similar to async interfacing with `ttnn::Tensor`

I just want us to be aware that with this PR we move async + the associated complexity into a core Metal Runtime data structure.

Metal Runtime was originally intended to be single threaded with no handoffs being done between main and worker threads (these are meant to be orchestrated by TTNN).

I bring this up because this may cause host side performance overheads when:
1. The user just wants to call Metal APIs directly and now has to go through the work executor path
2. TTNN APIs responsible for creating tensors now call `push_work` in the `Tensor::allocate()` function, which calls `push_work` in the buffer `allocate` function.

We should at least run performance regressions and get feedback from folks responsible for Metal Runtime architecture.

Reply:
> We should at least run performance regressions and get feedback from folks responsible for Metal Runtime architecture. 

Agreed! These were run and passed. Runs linked in PR description.

> I bring this up because this may cause host side performance overheads when:

Agreed! This is a good concern. After we discussed this approach with you, we thought deeply about performance implications. This change does not introduce any substantial overhead.

Reply:
Hey @ayerofieiev-tt yeah I was hoping to run this past them since this is more of an architectural change that adds more complexity to a core data structure in the Metal layer.